flink-table-store-core/src/main/java/org/apache/flink/table/store/file/manifest/ManifestFile.java

```diff
@@ -52,23 +55,32 @@ public class ManifestFile {
     private static final Logger LOG = LoggerFactory.getLogger(ManifestFile.class);
 
     private final RowType partitionType;
+    private final RowType entryType;
     private final ManifestEntrySerializer serializer;
+    private final FieldStatsArraySerializer statsSerializer;
     private final BulkFormat<RowData, FileSourceSplit> readerFactory;
     private final BulkWriter.Factory<RowData> writerFactory;
+    private final FileStatsExtractor fileStatsExtractor;
     private final FileStorePathFactory pathFactory;
     private final long suggestedFileSize;
 
     private ManifestFile(
             RowType partitionType,
+            RowType entryType,
```

Comment:
Can we just keep `ManifestFile` as it is?
I don't feel we need to introduce MetricFileWriter to it.

Reply:
The reason why we change this `ManifestFile` class is :  we've changed the `BaseFileWriter` to use the `FileWriter<T, Metric>.Factory` ( It use `BulkWriter.Factory<T>` before).  So we have to change the `ManifestEntryWriter` implementation so that we could still extend the `BaseFileWriter`.

Reply:
Although we don't rely on any metric which is generated from the `MetricFileWriter`

Reply:
Please see: https://github.com/apache/flink-table-store/pull/117/files#diff-4b9ebe9adecbf113c86f69fe4ef1c4fbb0395798eefe476e2d6f2de629e09e53R164-R166